### ISSUE TITLE
CBMC parser crashes due to a crash in CBMC

### ISSUE THREAD

**Issue by zhassan-aws** [model-checking/kani#861]
I tried this code:

```rust
fn main() {
    let a: Vec<Vec<i32>> = vec![vec![]; 1];
    assert!(false);
}
```

using the following command line invocation:

```
kani --unwind 2 test.rs
```

with Kani version: `a6118881e7f`

I expected to see this happen: Verification failed

Instead, this happened: 
```
$ kani --unwind 2 test.rs 
Traceback (most recent call last):
  File "/home/ubuntu/git/kani/scripts/kani.py", line 198, in run_cmd
    returncode = cbmc_json_parser.transform_cbmc_output(stdout, output_style)
  File "/home/ubuntu/git/kani/scripts/cbmc_json_parser.py", line 121, in transform_cbmc_output
    raise Exception("CBMC Crashed - Unable to present Result")
Exception: CBMC Crashed - Unable to present Result

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/git/kani/scripts/kani", line 201, in <module>
    sys.exit(main())
  File "/home/ubuntu/git/kani/scripts/kani", line 141, in main
    retcode = kani.run_cbmc(
  File "/home/ubuntu/git/kani/scripts/kani.py", line 346, in run_cbmc
    return run_cmd(
  File "/home/ubuntu/git/kani/scripts/kani.py", line 200, in run_cmd
    raise Exception("JSON Parsing Error")
Exception: JSON Parsing Error
```
Making this change to the parser:
```
$ git diff scripts/cbmc_json_parser.py
diff --git a/scripts/cbmc_json_parser.py b/scripts/cbmc_json_parser.py
index 3da07aa1851..ad3e5877045 100755
--- a/scripts/cbmc_json_parser.py
+++ b/scripts/cbmc_json_parser.py
@@ -118,6 +118,7 @@ def transform_cbmc_output(cbmc_response_string, output_style):
         # DynTrait tests generate a non json output due to "Invariant check failed" error
         # For these cases, we just produce the cbmc output unparsed
         # TODO: Parse these non json outputs from CBMC
+        print(cbmc_response_string)
         raise Exception("CBMC Crashed - Unable to present Result")
 
     return num_failed > 0
```
reveals the CBMC crash:
```
          }--- begin invariant violation report ---
Invariant check failed
File: ../src/util/expr_cast.h:256 function: validate_operands
Condition: allow_more ? value.operands().size()>=number : value.operands().size()==number
Reason: Unary expressions must have one operand
Backtrace:
cbmc(print_backtrace(std::ostream&)+0x50) [0x563a870be270]
cbmc(get_backtrace[abi:cxx11]()+0x169) [0x563a870be819]
cbmc(std::enable_if<std::is_base_of<invariant_failedt, invariant_failedt>::value, void>::type invariant_violated_structured<invariant_failedt, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x48) [0x563a86957368]
cbmc(validate_operands(exprt const&, unsigned long, char const*, bool)+0xd8) [0x563a86a49828]
cbmc(json(exprt const&, namespacet const&, dstringt const&)+0x4373) [0x563a86cecbe3]
cbmc(json(exprt const&, namespacet const&, dstringt const&)+0x2bbf) [0x563a86ceb42f]
cbmc(json(exprt const&, namespacet const&, dstringt const&)+0x2bbf) [0x563a86ceb42f]
cbmc(json(exprt const&, namespacet const&, dstringt const&)+0x2bbf) [0x563a86ceb42f]
cbmc(convert_decl(json_objectt&, conversion_dependenciest const&, trace_optionst const&)+0xf2c) [0x563a86cf5a5c]
cbmc(void convert<json_stream_arrayt>(namespacet const&, goto_tracet const&, json_stream_arrayt&, trace_optionst)+0x5c2) [0x563a8697a2d2]
cbmc(output_properties_with_traces(std::map<dstringt, property_infot, std::less<dstringt>, std::allocator<std::pair<dstringt const, property_infot> > > const&, goto_trace_storaget const&, trace_optionst const&, unsigned long, ui_message_handlert&)+0xbec) [0x563a86991b5c]
cbmc(all_properties_verifier_with_trace_storaget<multi_path_symex_checkert>::report()+0x94) [0x563a8695f3e4]
cbmc(cbmc_parse_optionst::doit()+0xf9a) [0x563a8695e81a]
cbmc(parse_options_baset::main()+0x8f) [0x563a8695547f]
cbmc(main+0x39) [0x563a86943d39]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7f09442f70b3]
cbmc(_start+0x2e) [0x563a86956d7e]


--- end invariant violation report ---

Traceback (most recent call last):
  File "/home/ubuntu/git/kani/scripts/kani.py", line 198, in run_cmd
```

**Comment by zhassan-aws** [comment]
Tracked on the CBMC side by https://github.com/diffblue/cbmc/issues/6691

**Comment by zhassan-aws** [comment]
This is fixed by the upgrade to CBMC 5.52.0 done in #928.
